Sources/HTTP/HTTPConnection.swift

```diff
@@ -0,0 +1,620 @@
+//
+//  HTTPConnection.swift
+//  HTTP
+//
+//  Created by Helge Hess on 22.10.17.
+//  Copyright © 2017 ZeeZide GmbH. All rights reserved.
+//
+
+import Foundation
+import Dispatch
+import CHTTPParser
+
+/// A connection to a single HTTP client.
+///
+/// The connection owns a queue which is used to synchronize all access to the
+/// API. Writers and `finishedProcessing` callbacks alike.
+///
+/// Back-pressure is supported. That is, until the body processor of a handler
+/// marks a chunk as processed, the connection suspends reading more data.
+/// Note that some reads might still be in-flight and will be buffered until
+/// the connection is resumed again.
+///
+/// Since HTTP clients can pipeline calls, a connection can run multiple
+/// handlers at the same time. To maintain proper response ordering,
+/// response-writer objects will be 'corked' until the responses ahead of them
+/// are completely written out.
+///
+internal class HTTPConnection : CustomStringConvertible {
+
+  /// The `HTTPServer` which created the connection. If the connection shuts
+  /// down, it will unregister from that server.
```

Comment:
Nit: spacing / color violations

Reply:
*color violations*??? :-)

2 spaces are right, has ARI approval: http://www.alwaysrightinstitute.com/swifter-space/

but feel free to change it.